src/state/roles/actions.js

```diff
@@ -100,3 +135,21 @@ export const sendDeleteRole = roleCode => dispatch =>
       });
     });
   });
+
+export const fetchUserRefs = (roleCode, page = { page: 1, pageSize: 10 }, params = '') => dispatch =>
+  new Promise((resolve) => {
+    dispatch(toggleLoading('references'));
+    getUserReferences(roleCode, page, params).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(setUserRefs(data.payload));
+          dispatch(setPage(data.metaData));
+          resolve();
```

Comment:
You can call `resolve()` only once, after the if/else block, after the last dispatch